lib/proxy_api/dhcp.rb

```diff
@@ -22,9 +22,12 @@ def subnet(subnet)
 
     def unused_ip(subnet, mac = nil)
       params = {}
-      params.merge!({:mac => mac}) if mac.present?
+      params[:mac] = mac if mac.present?
 
-      params.merge!({:from => subnet.from, :to => subnet.to}) if subnet.from.present? && subnet.to.present?
+      if subnet.from.present? && subnet.to.present?
+        params[:from] = subnet.from
+        params[:to] = subnet.to
+      end
```

Comment:
This might be reasonable case when to use `merge` tho, but I can live with that enforcing.